lib/timers/promises.js

```diff
@@ -122,7 +124,110 @@ function setImmediate(value, options = {}) {
   });
 }
 
+function setInterval(after, value, options = {}) {
+  const args = value !== undefined ? [value] : value;
```

Comment:
Nit: for validation prefer using `internal/validators`  (e.g. for validating an object you can call `validateObject`) etc